elytron/src/main/java/org/wildfly/extension/elytron/ModifiableRealmDecorator.java

```diff
@@ -148,152 +137,76 @@ protected void performRuntime(final OperationContext context, final ModelNode op
+            resourceRegistration.registerOperationHandler(new SimpleOperationDefinition(ElytronDescriptionConstants.REMOVE_IDENTITY, descriptionResolver, IDENTITY), new RemoveIdentityHandler());
         }
 
         @Override
         protected void executeRuntimeStep(final OperationContext context, final ModelNode operation) throws OperationFailedException {
+            String principalName = IDENTITY.resolveModelAttribute(context, operation).asString();
             ModifiableSecurityRealm modifiableRealm = getModifiableSecurityRealm(context);
-            String principalName = PathAddress.pathAddress(operation.get(ModelDescriptionConstants.ADDRESS)).getLastElement().getValue();
 
-            ModifiableRealmIdentity realmIdentity = null;
+            ModifiableRealmIdentity identity = null;
             try {
-                realmIdentity = modifiableRealm.getRealmIdentityForUpdate(new NamePrincipal(principalName));
+                identity = modifiableRealm.getRealmIdentityForUpdate(new NamePrincipal(principalName));
 
-                if (!realmIdentity.exists()) {
+                if (!identity.exists()) {
                     throw new OperationFailedException(ROOT_LOGGER.identityNotFound(principalName));
                 }
 
-                realmIdentity.delete();
-                realmIdentity.dispose();
+                identity.delete();
+                identity.dispose();
             } catch (RealmUnavailableException e) {
                 throw ROOT_LOGGER.couldNotCreateIdentity(principalName, e);
             } finally {
-                if (realmIdentity != null) {
-                    realmIdentity.dispose();
+                if (identity != null) {
+                    identity.dispose();
                 }
             }
         }
     }
 
-    static class ReadSecurityDomainIdentityHandler extends ElytronRuntimeOnlyHandler {
+    static class ReadIdentityHandler extends ElytronRuntimeOnlyHandler {
 
-        public static final SimpleAttributeDefinition NAME = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.NAME, ModelType.STRING, false)
+        static final SimpleAttributeDefinition IDENTITY = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.IDENTITY, ModelType.STRING, false)
                 .setAllowExpression(false)
                 .build();
 
         static void register(ManagementResourceRegistration resourceRegistration, ResourceDescriptionResolver descriptionResolver) {
-            resourceRegistration.registerOperationHandler(new SimpleOperationDefinition(ElytronDescriptionConstants.READ_IDENTITY, descriptionResolver, NAME), new ReadSecurityDomainIdentityHandler());
-        }
-
-        private ReadSecurityDomainIdentityHandler() {
+            resourceRegistration.registerOperationHandler(new SimpleOperationDefinition(ElytronDescriptionConstants.READ_IDENTITY, descriptionResolver, IDENTITY), new ReadIdentityHandler());
```

Comment:
This needs to use a builder and .setRuntimeOnly(). This comment applies to all these ops.